File: sample-projects/kotlin/app/src/main/kotlin/com/example/app/Application.kt
package com.example.app

import com.example.core.Result
import com.example.core.capitalizeWords
import com.example.core.truncate
import com.example.models.Address
import com.example.models.Permission
import com.example.models.User
import com.example.models.UserRepository

fun main() {
    val repository = UserRepository()
    val service = UserService(repository)

    val result = service.createUser("john doe", "john@example.com", Permission.ADMIN)
    when (result) {
        is Result.Success -> {
            val user = result.value
            println("Created user: ${user.name} (${service.describeIdentifiable(user)})")
            println("Valid: ${service.validateEntity<Unit>(user)}")
        }
        is Result.Failure -> println("Error: ${result.error}")
    }

    val address = Address(
        street = "123 Main St",
        city = "springfield",
        postalCode = "12345",
        country = "US"
    )
    println("City: ${address.city.capitalizeWords()}")
    println("Street: ${address.street.truncate(10)}")

    val admins = service.listAdmins()
    println("Admins: ${admins.size}")
}

File: sample-projects/ruby/lib/app/application.rb
# typed: strict
# frozen_string_literal: true

require_relative '../core/string_utils'
require_relative '../models/address'
require_relative '../models/permission'
require_relative 'user_service'

module App
  class Application
    extend T::Sig

    sig { void }
    def self.run
      repository = Models::UserRepository.new
      service = UserService.new(repository)

      result = service.create_user("john doe", "john@example.com", Models::Permission::Admin)
      if result.success?
        puts "User created successfully"
      else
        failure = T.cast(result, Core::Failure)
        puts "Error: #{failure.error}"
      end

      address = Models::Address.new(
        street: "123 Main St",
        city: "springfield",
        postal_code: "12345",
        country: "US"
      )
      puts "City: #{Core::StringUtils.capitalize_words(address.city)}"
      puts "Street: #{Core::StringUtils.truncate(address.street, 10)}"

      admins = service.list_admins
      puts "Admins: #{admins.length}"
    end
  end
end
